### ISSUE TITLE
Issues compiling on node 8.9.0

### ISSUE THREAD

**Issue by mmucklo** [brave/ad-block#78]
I'm trying to install ad-block on node 8.9.0 (the 8.9.0-slim image from docker hub):

I'm getting the following compilation-time error:

```
error /opt/preprocessor/node_modules/ad-block: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: /opt/preprocessor/node_modules/ad-block
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@8.9.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/opt/preprocessor/node_modules/ad-block/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.node-gyp/8.9.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.node-gyp/8.9.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.node-gyp/8.9.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/opt/preprocessor/node_modules/ad-block',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/opt/preprocessor/node_modules/ad-block/build'
  CXX(target) Release/obj.target/ad-block/addon.o
  CXX(target) Release/obj.target/ad-block/ad_block_client_wrap.o
../ad_block_client_wrap.cc: In function 'v8::Local<v8::Object> ad_block_client_wrap::ToLocalObject(v8::Isolate*, FilterList)':
../ad_block_client_wrap.cc:52:3: error: 'for_each' is not a member of 'std'
   std::for_each(filter_list.langs.begin(), filter_list.langs.end(),
   ^
../ad_block_client_wrap.cc: In function 'v8::Local<v8::Array> ad_block_client_wrap::ToLocalObject(v8::Isolate*, const std::vector<FilterList>&)':
../ad_block_client_wrap.cc:67:3: error: 'for_each' is not a member of 'std'
   std::for_each(list.begin(), list.end(),
   ^
../ad_block_client_wrap.cc: In static member function 'static void ad_block_client_wrap::AdBlockClientWrap::New(const v8::FunctionCallbackInfo<v8::Value>&)':
../ad_block_client_wrap.cc:168:59: warning: 'v8::Local<v8::Object> v8::Function::NewInstance(int, v8::Local<v8::Value>*) const' is deprecated (declared at /root/.node-gyp/8.9.0/include/node/v8.h:3831): Use maybe version [-Wdeprecated-declarations]
     args.GetReturnValue().Set(cons->NewInstance(argc, argv));
                                                           ^
ad-block.target.mk:108: recipe for target 'Release/obj.target/ad-block/ad_block_client_wrap.o' failed
make: Leaving directory '/opt/preprocessor/node_modules/ad-block/build'
make: *** [Release/obj.target/ad-block/ad_block_client_wrap.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.9.49-moby
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/preprocessor/node_modules/ad-block
gyp ERR! node -v v8.9.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by mmucklo** [comment]
Update:

yarn add https://github.com/brave/ad-block.git#b74f508be52931604199195905d3113bf450c851
(latest commit as of this writing)

Does a little better, however I still get the following:

```
error /opt/preprocessor/node_modules/ad-block: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: /opt/preprocessor/node_modules/ad-block
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@8.9.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/opt/preprocessor/node_modules/ad-block/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/opt/preprocessor/node_modules/ad-block/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/opt/preprocessor/node_modules/ad-block/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.node-gyp/8.9.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.node-gyp/8.9.0',
gyp info spawn args   '-Dnode_gyp_dir=/opt/preprocessor/node_modules/ad-block/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.node-gyp/8.9.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/opt/preprocessor/node_modules/ad-block',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/opt/preprocessor/node_modules/ad-block/build'
  CXX(target) Release/obj.target/ad-block/addon.o
In file included from ../addon.cc:6:0:
/root/.node-gyp/8.9.0/include/node/node.h:475:43: error: overloaded function with no contextual type information
       (node::addon_register_func) (regfunc),                          \
                                           ^
/root/.node-gyp/8.9.0/include/node/node.h:506:3: note: in expansion of macro 'NODE_MODULE_X'
   NODE_MODULE_X(modname, regfunc, NULL, 0)
   ^
../addon.cc:28:1: note: in expansion of macro 'NODE_MODULE'
 NODE_MODULE(ad_block, InitAll)
 ^
../addon.cc:18:6: warning: 'void {anonymous}::InitAll(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)' defined but not used [-Wunused-function]
 void InitAll(v8::Local<v8::Object> exports, v8::Local<v8::Value> unused,
      ^
ad-block.target.mk:108: recipe for target 'Release/obj.target/ad-block/addon.o' failed
make: Leaving directory '/opt/preprocessor/node_modules/ad-block/build'
make: *** [Release/obj.target/ad-block/addon.o] Error 1
gyp ERR! build error
```

**Comment by mmucklo** [comment]
Fixed as of 3.1.10
